### PR TITLE
Combine team battles

### PR DESCRIPTION
**Pull request by budokaiman**
Include team battles in "my battles/retros/storyboards". I opted to include team battles in the existing list as I found it the easiest to navigate as well as making logical sense (especially if using retros/storyboards). If a user is added to an existing team, they will be able to see past battles and retros for the team, but they will still be listed under current ones. There are no duplicates or need for condensing results. Team items are indicated by the team name next to the item name.

I had also considered the following before settling on the combined view
* showing team battles underneath my battles - this make the landing page look bloated and there were frequently duplicated results. It also meant having to look around more when wishing to join a new team battle
* showing team battles between my battles and create - this made the landing page very squished, though easier to navigate
* automatically adding team members to battles on battle creation - this has the downside of not really seeing who is participating, and historical battles still need to be found through individual teams

image

Addresses Issue #168

### CODE REVIEW

File: ui/src/i18n/i18n-types.ts [hunk 1]
```diff
@@ -994,10 +994,18 @@ type RootTranslation = {
    * M​y​ ​R​e​t​r​o​s
    */
   myRetros: string;
+  /**
+   * M​y​ ​R​e​t​r​o​s
```
Comment (scottux) [inline ui/src/i18n/i18n-types.ts:998]:
```suggestion
   * T​eam​ ​R​e​t​r​o​s
```

File: ui/src/i18n/i18n-types.ts [hunk 2]
```diff
@@ -994,10 +994,18 @@ type RootTranslation = {
...
+   */
+  teamRetros: string;
   /**
    * M​y​ ​S​t​o​r​y​b​o​a​r​d​s
    */
   myStoryboards: string;
+  /**
+   * M​y​ ​S​t​o​r​y​b​o​a​r​d​s
```
Comment (scottux) [inline ui/src/i18n/i18n-types.ts:1006]:
```suggestion
   * Team​ ​S​t​o​r​y​b​o​a​r​d​s
```

**Review by scottux — CHANGES_REQUESTED** [review]
Just two comment updates that I see. I have made suggestions.

**Comment by scottux** [comment]
I question the naming of `ownerNameField` and `showOwnerName`. The default seems to be showing the team name, but functionally it could be any extra field that is defined.

Maybe it should be more explicit `teamNameField` and `showTeamName` instead.

**Comment by budokaiman** [comment]
> I question the naming of `ownerNameField` and `showOwnerName`. The default seems to be showing the team name, but functionally it could be any extra field that is defined.
> 
> Maybe it should be more explicit `teamNameField` and `showTeamName` instead.

I went back and forth on this, and decided to stick with "owner name field" in case there was ever a time that an owning user should be shown over the owning team. This would reduce the need for extra variables or a subsequent rename. Let me know if you still disagree, I can change the naming

File: ui/src/i18n/en/index.ts
Comment (StevenWeathers) [inline ui/src/i18n/en/index.ts:1]: I see new additions to the locale files however these new keys are not in use in this PR, either remove them or use them.

**Comment by StevenWeathers** [comment]
I greatly appreciate this PR and will be happy to accept it once the locale file issue is sorted out.

File: ui/src/i18n/en/index.ts
Comment (budokaiman) [reply ui/src/i18n/en/index.ts:1]: This was definitely an artefact of when I was trying out the double columns. Removed.

**Comment by budokaiman** [comment]
> I greatly appreciate this PR and will be happy to accept it once the locale file issue is sorted out.

Cheers, I greatly appreciate the software you've built, so happy to contribute where able.

**Review by scottux — APPROVED** [review]

**Review by StevenWeathers — APPROVED** [review]

**Pull request merged**